### PR TITLE
Add Orbital Camera component

### PR DESCRIPTION
**Pull request by sorskoot**
Add an orbital camera component that can be used with a mouse and touch. The component can be added to the `nonVRCamera` instead of the `mouseLook` component.

### CODE REVIEW

File: orbital-camera.ts [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+            this.updateCamera();
+        }
+    }
+
+    private updateCamera() {
+        this.object.setPositionWorld([
+            this.radial * Math.sin(deg2rad(this.azimuth)) * Math.cos(deg2rad(this.polar)),
+            this.radial * Math.sin(deg2rad(this.polar)),
+            this.radial * Math.cos(deg2rad(this.azimuth)) * Math.cos(deg2rad(this.polar)),
+        ]);
```
Comment (Squareys) [inline orbital-camera.ts:129]: This is creating a temporary minimal garbage list with the `[]` literal.

You could move that into a temp vector, and fill it before passing to setPositionWorld:

```js
tempVec[0] = ...;
tempVec[1] = ...;
tempVec[2] = ...;
setPositionWorld(tempVec);
```

File: tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "target": "ES2022",
-    "module": "ES6",
+    "module": "NodeNext",
```
Comment (Squareys) [inline tsconfig.json:4]: Was this required? 🤔

Comment (sorskoot) [reply tsconfig.json:4]: Yes, it gives an error by the TypeScript compiler.

```
tsconfig.json:4:15 - error TS5110: Option 'module' must be set to 'NodeNext' when option 'moduleResolution' is set to 'NodeNext'.
4     "module": "ES6",
```

File: orbital-camera.ts (hunk 1 above)
Comment (sorskoot) [reply orbital-camera.ts:129]: Fixed.

File: orbital-camera.ts [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    @property.float(0.5)
+    ySensitivity = 0.5;
+
+    @property.float(0.02)
+    zoomSensitivity = 0.02;
+
+    @property.float(0.9)
+    decelerationFactor = 0.9;
+
+    private mouseDown: boolean = false;
```
Comment (DavidPeicho) [inline orbital-camera.ts:51]: Missing trailing `_` I guess, except if the user can change that anytime in js, but doesn't seem like a public api

File: orbital-camera.ts [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+    xSensitivity = 0.5;
+
+    @property.float(0.5)
+    ySensitivity = 0.5;
+
+    @property.float(0.02)
+    zoomSensitivity = 0.02;
+
+    @property.float(0.9)
+    decelerationFactor = 0.9;
```
Comment (DavidPeicho) [inline orbital-camera.ts:49]: That's more commonly referred as `damping` I think

File: orbital-camera.ts [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+    start(): void {
+        this.object.getPositionWorld(this.origin);
+        this.updateCamera();
+    }
+
+    onActivate(): void {
+        const canvas = this.engine.canvas;
+
+        canvas.addEventListener('mousemove', this.onMouseMove);
```
Comment (DavidPeicho) [inline orbital-camera.ts:71]: Post-merge: Would be nice to allow choosing the element itself for inputs.

File: orbital-camera.ts [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+        this.touchStartX = 0;
+        this.touchStartY = 0;
+
+        this.azimuthSpeed = 0;
+        this.polarSpeed = 0;
+    }
+
+    update(): void {
+        if (!this.mouseDown) {
+            // Apply deceleration only when the user is not actively dragging
```
Comment (DavidPeicho) [inline orbital-camera.ts:111]: We use `/* */` everywhere, a bit different than most codebase I guess but now we are too deep in :D

File: orbital-camera.ts [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+
+        canvas.addEventListener('mousemove', this.onMouseMove);
+        if (this.mouseButtonIndex === 2) {
+            canvas.addEventListener('contextmenu', preventDefault, {passive: false});
+        }
+        canvas.addEventListener('mousedown', this.onMouseDown);
+        canvas.addEventListener('mouseup', this.onMouseUp);
+        canvas.addEventListener('wheel', this.onMouseScroll, {passive: false});
+
+        canvas.addEventListener('touchstart', this.onTouchStart, {passive: false});
```
Comment (DavidPeicho) [inline orbital-camera.ts:79]: Wouldn't it be bettee to use the pointer API instead? So it's somehow unified mouse+touch just with an extra pointer

File: orbital-camera.ts [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+            this.mouseDown = true; // Treat touch like mouse down
+        } else if (e.touches.length === 2) {
+            // Calculate initial pinch distance
+            this.initialPinchDistance = this.getDistanceBetweenTouches(e.touches);
+            e.preventDefault(); // Prevent default pinch actions
+        }
+    };
+
+    private onTouchMove = (e: TouchEvent) => {
+        if (this.active && this.mouseDown) {
```
Comment (DavidPeicho) [inline orbital-camera.ts:205]: Early return

File: orbital-camera.ts (hunk 4 above)
Comment (sorskoot) [reply orbital-camera.ts:49]: I agree.

File: orbital-camera.ts (hunk 6 above)
Comment (sorskoot) [reply orbital-camera.ts]: I'm not a fan, but I changed it for consistency. I'll look into adding ESLint rules to enforce these kinds of rules in future.

**Pull request merged**